src/core_plugins/input_control_vis/public/components/vis/list_control.js

```diff
@@ -1,77 +1,71 @@
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-import Select from 'react-select';
 import { FormRow } from './form_row';
 
 import {
   EuiFieldText,
+  EuiComboBox,
 } from '@elastic/eui';
 
 export class ListControl extends Component {
-  constructor(props) {
-    super(props);
 
-    this.handleOnChange = this.handleOnChange.bind(this);
-    this.truncate = this.truncate.bind(this);
-  }
-
-  handleOnChange(evt) {
-    let newValue = '';
-    if (evt) {
-      newValue = evt;
-    }
-    this.props.stageFilter(this.props.controlIndex, newValue);
-  }
-
-  truncate(selected) {
-    if (selected.label.length <= 24) {
-      return selected.label;
-    }
-    return `${selected.label.substring(0, 23)}...`;
+  handleOnChange = (selectedOptions) => {
+    this.props.stageFilter(this.props.controlIndex, selectedOptions);
   }
 
   renderControl() {
-    if (!this.props.control.isEnabled()) {
-      // react-select clobbers the tooltip, so just returning a disabled input instead
+    if (this.props.disableMsg) {
       return (
         <EuiFieldText
+          placeholder="Select..."
           disabled={true}
         />
       );
     }
 
+    const options = this.props.options.map(option => {
+      option['data-test-subj'] = `option_${option.value.replace(' ', '_')}`;
+      return option;
```

Comment:
This is going to happen every time render happens, but it is mutating the props. It should probably be done elsewhere or else not mutate props, I think. I've been bitten often by surprising mutations in past projects, so I'm a bit of a stickler about this. Probably not super important in this case, but...

Reply:
where would you recommend moving the logic?

Reply:
Maybe something like this: https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
I just change the map function to create a new object instead of returning the object from props. That way, props is not getting modified